vita3k/io/src/FilesystemHandlers.cpp

```diff
@@ -0,0 +1,48 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include "io/FilesytemHandlers.h"
+
+/**
+ * Static methods should be defined outside the class.
+ */
+
+FilesystemHandlers *FilesystemHandlers::my_instance = nullptr;
```

Comment:
what is this static variable for? :D

Reply:
I am creating a [Singleton](https://en.wikipedia.org/wiki/Singleton_pattern) for the file handler. The reson why I am not just doing a namespace or just functions is that later I will need to save the states of certain objects which is much simpler to do in a singular object itself.

Reply:
Ahh yeah I've seen it before. I suggest storing the object in [`HostState`](https://github.com/Vita3K/Vita3K/blob/master/vita3k/host/include/host/state.h) and grabbing it from there when you need it. That's how we've traditionally gone about it, and it will probably make resetting the console + switching games easier to perform if this isn't a static/global object (just `host = HostState()` vs. `FilesystemHandlers::reset_instance(); host = HostState()` if that's fair).

Reply:
Right now I don't see the need of a reset instance, as the abstraction is to interface with the files, and not its content. Also, with the way I am currently doing it, I am making sure that there will always only be one instance of the handler, while with Host i will not be guaranteed. Tho I will look into what you are proposing.